File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        return self.bstToList(root, [])

    def bstToList(self, root: Optional[TreeNode], memoCache) -> List[int]:
        if not root:
            return memoCache

        self.bstToList(root.left, memoCache)

        memoCache.append(root.val)

        self.bstToList(root.right, memoCache)

        return memoCache

        

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution
{
	public List<int> RightSideView(TreeNode root)
	{
		List<int> output = new List<int>();
		
		if(root == null)
			return output;
		
		Queue<TreeNode> queue = new Queue<TreeNode>();
		queue.Enqueue(root);
			
		while(queue.Count > 0)
		{
			int levelSize = queue.Count();
			
			for (int i = 0; i < levelSize; i++)
			{
				var current = queue.Dequeue();
				if (i == levelSize - 1)
					output.Add(current.val);

				if(current.left != null)
					queue.Enqueue(current.left);

				if(current.right != null)
					queue.Enqueue(current.right);
			}
		}	
		
		return output;
	}
}